Library/Homebrew/readall.rb

```diff
@@ -0,0 +1,79 @@
+require "formula"
+require "tap"
+require "thread"
+require "readall"
+
+module Readall
+  class << self
+    def valid_ruby_syntax?(ruby_files)
+      failed = false
+      workers = (0...Hardware::CPU.cores).map do
+        Thread.new do
+          begin
+            while rb = ruby_files.pop(true)
+              # As a side effect, print syntax errors/warnings to `$stderr`.
+              failed = true if syntax_errors_or_warnings?(rb)
+            end
+          rescue ThreadError # ignore empty queue error
+          end
+        end
+      end
+      workers.each(&:join)
+      !failed
+    end
+
+    def valid_aliases?(alias_dirs)
+      failed = false
+      alias_dirs.each do |alias_dir|
+        next unless alias_dir.directory?
+        Pathname.glob("#{alias_dir}/*").each do |f|
+          next unless f.symlink?
+          next if f.file?
+          onoe "Broken alias: #{f}"
+          failed ||= true
+        end
+      end
+      !failed
+    end
+
+    def valid_formulae?(formulae)
+      failed = false
+      formulae.each do |file|
+        begin
+          Formulary.factory(file)
+        rescue Interrupt
+          raise
+        rescue Exception => e
+          onoe "problem in #{file}"
```

Comment:
The preceding method `valid_aliases?` uses `onoe "Broken alias: #{f}"`. Would you mind adjusting the styling of the output a bit, e.g. by using:

```ruby
onoe "Problem in: #{file}"
```